packages/tokens/non-fungible/src/extensions/mintable/mod.rs

```diff
@@ -0,0 +1,81 @@
+mod storage;
+pub use self::storage::mint;
+
+mod test;
+
+use soroban_sdk::{contractclient, symbol_short, Address, Env};
+
+/// Mintable Trait for Non-Fungible Token
+///
+/// The `NonFungibleMintable` trait extends the `NonFungibleToken` trait to
+/// provide the capability to mint tokens. This trait is designed to be used in
+/// conjunction with the `NonFungibleToken` trait.
+///
+/// Excluding the `mint` functionality from the
+/// [`crate::non_fungible::NonFungibleToken`] trait is a deliberate design
+/// choice to accommodate flexibility and customization for various smart
+/// contract use cases.
+#[contractclient(name = "NonFungibleMintableClient")]
+pub trait NonFungibleMintable {
+    /// Creates a token with the next available `token_id` and assigns it to
+    /// `to`. Returns the `token_id` for the newly minted token.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `to` - The address receiving the new token.
+    ///
+    /// # Errors
+    ///
+    /// * [`crate::NonFungibleTokenError::MathOverflow`] - When all the
```

Comment:
what about a better name matching the reason?

Reply:
sure, why not :)